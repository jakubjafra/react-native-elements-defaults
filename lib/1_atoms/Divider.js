const _ = require('lodash');
const utils = require('../utils');

// TODO: Make this function immutable.
function override(defaultStyles) {
    const __oldModule__ = require('react-native-elements/src/divider/Divider');
    const __oldModuleDefault__ = __oldModule__.default;

    const Divider = function (props) {
        const newProps = {};

        newProps.style = utils.overrideTop(props.style, defaultStyles.style);

        return __oldModuleDefault__.call(undefined, _.defaultsDeep(newProps, props));
    };

    Divider.propTypes = __oldModuleDefault__.propTypes;
    __oldModule__.default = Divider;
}

module.exports = override;
