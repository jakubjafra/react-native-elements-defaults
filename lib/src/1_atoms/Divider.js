const _defaults = require('lodash.defaults');
const utils = require('../utils');

function compose(defaultStyles) {
    return function(__oldModuleDefault__) {
        const Divider = function(props) {
            const newProps = {};

            newProps.style = utils.overrideBottom(props.style, defaultStyles.style);

            return __oldModuleDefault__.call(undefined, _defaults(newProps, props));
        };

        Divider.propTypes = __oldModuleDefault__.propTypes;
        Divider.defaultProps = __oldModuleDefault__.defaultProps;

        return Divider;
    }
}

module.exports = {
    compose,
    override: new utils.ModuleOverride('atoms/divider', require('react-native-elements/src/divider/Divider'), compose),
};
