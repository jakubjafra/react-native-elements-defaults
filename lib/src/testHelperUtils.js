import 'react-native';
import React from 'react';
import renderer from "react-test-renderer";

const diff = require('jest-diff');

function defaultTests(name,
                      __oldElement__,
                      __element__,
                      basicTheme,
                      createElement = element => React.createElement(element, null, null)) {
    describe('with no theme applied', () => {
        let OldElement, Element;

        beforeAll(() => {
            let defaultStyles = {};

            OldElement = __oldElement__.default;
            Element = __element__.compose(defaultStyles)(OldElement);
        });

        test(`Old${name} and ${name} should be different`, () => {
            expect(OldElement).not.toBe(Element);
        });

        test(`Old${name} renders correctly`, () => {
            const tree = renderer.create(createElement(OldElement)).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`${name} renders correctly`, () => {
            const tree = renderer.create(createElement(Element)).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`Old${name} and ${name} returns the same thing if no theme is applied`, () => {
            const oldTree = renderer.create(createElement(OldElement)).toJSON();
            const tree = renderer.create(createElement(Element)).toJSON();

            expect(diff(tree, oldTree)).toMatchSnapshot();
        });
    });

    describe('with basic theme applied', () => {
        let OldElement, Element;

        beforeAll(() => {
            let defaultStyles = basicTheme;

            OldElement = __oldElement__.default;
            Element = __element__.compose(defaultStyles)(OldElement);
        });

        test(`Old${name} and ${name} should be different`, () => {
            expect(OldElement).not.toBe(Element);
        });

        test(`Old${name} renders correctly`, () => {
            const tree = renderer.create(createElement(OldElement)).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`${name} renders correctly`, () => {
            const tree = renderer.create(createElement(Element)).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`Old${name} and ${name} should be different caused by theme applied`, () => {
            const oldTree = renderer.create(createElement(OldElement)).toJSON();
            const tree = renderer.create(createElement(Element)).toJSON();

            expect(oldTree).not.toEqual(tree);
        });
    });
}

module.exports = {
    defaultTests
};
