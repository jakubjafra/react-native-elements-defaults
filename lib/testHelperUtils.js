import 'react-native';
import React from 'react';
import renderer from "react-test-renderer";

const diff = require('jest-diff');

function defaultTests(name, __oldElement__, __element__, basicTheme){
    describe('with no theme applied', () => {
        let OldElement, Element;

        beforeAll(() => {
            let defaultStyles = {};

            OldElement = __oldElement__.default;
            Element = __element__.compose(defaultStyles)(OldElement);
        });

        test(`Old${name} and ${name} should be different`, () => {
            expect(OldElement).not.toBe(Element);
        });

        test(`Old${name} renders correctly`, () => {
            const tree = renderer.create(<OldElement />).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`${name} renders correctly`, () => {
            const tree = renderer.create(<Element />).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`Old${name} and ${name} returns the same thing if no theme is applied`, () => {
            const oldTree = renderer.create(<OldElement />).toJSON();
            const tree = renderer.create(<Element />).toJSON();

            expect(diff(tree, oldTree)).toMatchSnapshot();
        });
    });

    describe('with basic theme applied', () => {
        let OldElement, Element;

        beforeAll(() => {
            let defaultStyles = basicTheme;

            OldElement = __oldElement__.default;
            Element = __element__.compose(defaultStyles)(OldElement);
        });

        test(`Old${name} and ${name} should be different`, () => {
            expect(OldElement).not.toBe(Element);
        });

        test(`Old${name} renders correctly`, () => {
            const tree = renderer.create(<OldElement />).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`${name} renders correctly`, () => {
            const tree = renderer.create(<Element />).toJSON();
            expect(tree).toMatchSnapshot();
        });

        test(`Old${name} and ${name} should be different caused by theme applied`, () => {
            const oldTree = renderer.create(<OldElement />).toJSON();
            const tree = renderer.create(<Element />).toJSON();

            expect(oldTree).not.toEqual(tree);
        });
    });
}

module.exports = {
    defaultTests
};
