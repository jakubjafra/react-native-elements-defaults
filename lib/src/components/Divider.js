const utils = require('../utils');
const functionalComponentCompose = require('../compose/functionalComponent');

const COMPOSE_NAME = 'Divider';
const compose = functionalComponentCompose.construct(COMPOSE_NAME);

module.exports = {
    name: COMPOSE_NAME,
    compose,
    override: new utils.ModuleOverride(
        COMPOSE_NAME,
        require('react-native-elements/src/divider/Divider'),
        compose,
        true,
    ),
};
