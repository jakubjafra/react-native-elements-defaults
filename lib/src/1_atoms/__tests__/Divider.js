import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import * as utils from '../../testHelperUtils';

describe('Divider', () => {
    utils.defaultTests(
        'Divider',
        require('react-native-elements/src/divider/Divider'),
        require('../Divider'),
        {
            style: {
                backgroundColor: 'red',
            }
        }
    );
});
