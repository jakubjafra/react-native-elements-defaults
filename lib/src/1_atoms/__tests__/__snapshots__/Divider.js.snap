// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider with basic theme applied Divider renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#e1e8ee",
        "height": 1,
      },
      Object {
        "backgroundColor": "red",
      },
    ]
  }
/>
`;

exports[`Divider with basic theme applied OldDivider renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#e1e8ee",
        "height": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`Divider with no theme applied Divider renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#e1e8ee",
        "height": 1,
      },
      undefined,
    ]
  }
/>
`;

exports[`Divider with no theme applied OldDivider and Divider returns the same thing if no theme is applied 1`] = `"[2mCompared values have no visual difference.[22m"`;

exports[`Divider with no theme applied OldDivider renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#e1e8ee",
        "height": 1,
      },
      undefined,
    ]
  }
/>
`;
